tutorials/tutorial-forecasting.md

```diff
@@ -0,0 +1,773 @@
+seven days in a week). Since we expect the seasonal variations 
+to be fairly constant over time, we use the additive method rather 
+than the multiplicative method, which is specified by the `trend` 
+and `seasonal` parameters.
+
+```python
+from statsmodels.tsa.api import ExponentialSmoothing
+fit = ExponentialSmoothing(np.asarray(ride_length_train['trip_length']), seasonal_periods = 56, trend = 'add', seasonal = 'add').fit()
+```
+
+We use the model that has been trained to make a forecast and compare 
+with the testing dataset.
+
+```python
+ride_length_test['forecast'] = fit.forecast(len(ride_length_test))
+```
+
+Now `ride_length_test` has a column with the observed values and 
+predicted values from January 22nd to January 31st. We can plot 
+these values on top of each other to make a visual comparison:
+
+```python
+import matplotlib.pyplot as plt
+plt.plot(ride_length_test)
+plt.title('Taxicab Ride Length from Financial District to Times Square by Time')
+plt.xlabel('Date')
+plt.ylabel('Ride Length (seconds)')
+plt.legend(['Observed', 'Predicted'])
+plt.show()
+```
+
+![alt text][rides_length_graph]
+
+The model predicts that the length of a trip from the Financial 
+District to Times Square fluctuates roughly between 16 minutes 
+and 38 minutes, with high points midday and low points overnight. 
+The trip length is notably longer during weekdays than it is 
+during weekends (January 23rd 24th, 30th, 31st). 
+
+The initial reaction from the plotted graph is that the model 
+does a relatively good job in capturing the overall trend, but at 
+times has quite a large margin of error. This can be due to the 
+inherent irregularity of Manhattan’s traffic situation with 
+frequent roadblocks, accidents and unexpected weather conditions. 
+Moreover, as it was the case with taxi pickup counts in our analysis 
+with R using the seasonal ARIMA model, the Holt-Winters model was 
+also thrown off by the anomalous data points on Martin Luther King 
+day on the previous Monday.
+
+### Takeaways from Analysis
+We looked at different ways you can build statistical models to 
+analyze time-series data and how you can leverage the full power 
+of the PostgreSQL ecosystem with TimescaleDB. In this tutorial 
+we looked at integrating TimescaleDB with R, Apache MADlib, and 
+Python. You can simply choose the option you are most familiar
+with from a vast number of choices that TimescaleDB inherits 
+from PostgreSQL. ARIMA and Holt-Winters are just a couple from a 
+wide variety of statistical models and machine learning algorithms 
+that you can use to analyze and make predictions on time-series 
+data in your TimescaleDB database. 
+
+[hello_nyc]: http://docs.timescale.com/latest/tutorials/tutorial-hello-nyc
+[install]: /getting-started/installation
+[install_r]: https://www.r-project.org/
+[install_python]: https://www.python.org/downloads/
+[gap_filling]: /using-timescaledb/reading-data#advanced-analytics
```

Comment:
You can actually use a more specific anchor here: `/using-timescaledb/reading-data#gap-filling`